src/redux/missions/missionSlice.jsx

```diff
@@ -0,0 +1,69 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const missionsUrl = 'https://api.spacexdata.com/v3/missions';
+
+const initialState = {
+  missions: [],
+  isLoading: false,
+  error: null,
+};
+
+export const fetchMissions = createAsyncThunk('fetchMissions', async () => {
+  try {
+    const response = await axios.get(missionsUrl);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
+
+const missionSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    joinMission: (state, action) => {
+      const mission = state.missions.find(
+        (item) => item.mission_id === action.payload,
+      );
+      mission.member = true;
+    },
+    leaveMission: (state, action) => {
+      const mission = state.missions.find(
+        (item) => item.mission_id === action.payload,
+      );
+      mission.member = false;
+    },
+  },
+  extraReducers(builder) {
+    builder
+      .addCase(fetchMissions.pending, (state) => ({
+        ...state,
+        isLoading: true,
+      }))
+      .addCase(
+        fetchMissions.fulfilled,
+        (state, action) => ({
+          ...state,
+          missions: action.payload,
+          isLoading: false,
+        }),
+
+        // state.missions = action.payload;
+        // state.isLoading = false;
```

Comment:
- [ ] Leaving unused code as comments make our codebase look messy. You can use git history if you need to revisit old pieces of code. Kindly remove these types of inline comments to keep your repository as professional as possible. 🙂️